stile/sys_tests.py

```diff
@@ -891,13 +891,392 @@ def __call__(self, array, linewidth = 0.01, scale = None, figsize = None,
+        @param xlabel        The x-axis label.
+                             [Default: None, meaning do not show a label for the x-axis]
+        @param ylabel        The y-axis label.
+                             [Default: None, meaning do not show a label for the y-axis]
+        @param xlim          Limits of x-axis (min, max).
+                             [Default: None, meaning do not set any limits for x]
+        @param ylim          Limits of y-axis (min, max).
+                             [Default: None, meaning do not set any limits for y]
+        @param hide_x        Whether hide the labels for x-axis.
+                             [Default: hide_x = False]
+        @param hide_y        Whether hide the labels for y-axis.
+                             [Default: hide_y = False]
+        @param alpha         0.0 transparent through 1.0 opaque
+                             [Default: alpha = 1.0]
+        @param linewidth     With of the vertical lines
+                             [Default: linewidth = 2.0]
+        @param text          Text to put on the figure
+                             [Default: None]
+        @param text_x        The X-coordinate of the text on the plot
+                             [Default: text_x = 0.9]
+        @param text_y        The Y-coordinate of the test on the plot
+                             [Default: text_y = 0.9]
+        @param fontsize      Font size of the text
+                             [Default: fontsize = 12]
+        @param vlines        Locations to plot vertical lines to indicate interesting
+                             values.
+                             [Default: None]
+        @param vcolor        Color or a list of color for vertical lines to plot.
+                             [Default: 'k']
+
+        @returns a matplotlib.figure.Figure object.
+        """
+
+        ## Define the plot
+        hist = plt.figure(figsize=figsize)
+        ax   = hist.add_subplot(1, 1, 1)
+
+        data_dim = len(data_list)
+        for ii in range(data_dim):
+
+            if type(data_list[0]) is list or type(data_list[0]) is numpy.ndarray:
+                multihist = True
+                data = data_list[ii]
+            else:
+                multihist = False
+                data = data_list
+
+            # mask data with NaN
+            data = data[numpy.isnan(data) == False]
+            data = numpy.asarray(data)
+
+            # trim the data if necessary
+            if limits is not None:
+                data = data[(data >= limits[0]) & (data <= limits[1])]
+
+            # decide which bin style to use
+            style_use = self.get_param_value(self, binning_style, ii, data_dim,
+                                             multihist=multihist)
+
+            # now support constant bin size, Scott rule, and Freedman rule
+            if style_use in ['scott', 'freedman', 'manual']:
+                if (style_use is 1):
```

Comment:
I think this line (and the next few) need to change to the strings instead of integers version of the style keywords, eg `if (style_use is 'scott')`.